src/CustomControls.Shared/ModernMapPanel.cs

```diff
@@ -0,0 +1,627 @@
+            FrameworkElement bottomLeftAccessory = null;
+
+            foreach (UIElement element in Children)
+            {
+                if (element is UIElement child)
+                {
+                    switch (GetRole(child))
+                    {
+                        case MapRole.Titlebar:
+                            titlebar = child;
+                            break;
+                        case MapRole.GeoView:
+                            geoview = child;
+                            break;
+                        case MapRole.ContextCanvas:
+                            canvas = child;
+                            break;
+                        case MapRole.Attribution:
+                            attribution = child;
+                            break;
+                        case MapRole.CardAppendage:
+                            appendage = child;
+                            break;
+                        case MapRole.Card:
+                            if (child != TopCard)
+                            {
+                                // TODO - determine if this is the best way
+                                child.Arrange(new Rect(0, 0, 0, 0));
+                            }
+
+                            break;
+                        case MapRole.Accessory:
+                            if (child is FrameworkElement fe)
+                            {
+                                if (fe.HorizontalAlignment == HorizontalAlignment.Right)
+                                {
+                                    if (fe.VerticalAlignment == VerticalAlignment.Top)
+                                    {
+                                        topRightAccessory = fe;
+                                    }
+                                    else
+                                    {
+                                        bottomRightAccessory = fe;
+                                    }
+                                }
+                                else
+                                {
+                                    if (fe.VerticalAlignment == VerticalAlignment.Top)
+                                    {
+                                        topLeftAccessory = fe;
+                                    }
+                                    else
+                                    {
+                                        bottomLeftAccessory = fe;
+                                    }
+                                }
+                            }
+
+                            break;
+                        case MapRole.ModalLightbox:
+                            lightBoxes.Add(child);
+                            break;
+                        default:
+                            child.Arrange(new Rect(0, 0, 0, 0));
+                            break;
+                    }
+                }
+            }
+
+            double titleBarBottomY = 0;
+            double cardTopY = 0;
```

Comment:
This assignment is unnessecary, the value is assigned again at line 422 before being used. Its more readable this way, but VS does warn about this (greyed out text)